lib/layout/data_grid.dart

```diff
@@ -159,7 +163,19 @@ class EnsembleDataRow extends StatefulWidget
   @override
   Map<String, Function> setters() {
     return {
-      'visible': (dynamic v) => visible = Utils.getBool(v, fallback: visible)
+      'visible': (dynamic v) {
+        final bool newValue = Utils.getBool(v, fallback: visible);
+
+        // checking if old value and new value are different
+        if (visible != newValue) {
+          visible = newValue;
+
+          if (onVisibilityChanged != null) {
```

Comment:
don't need the null check. The ? in the statement already checks for that. Dart is pretty cool :-)